what-is-mtls.md

```diff
@@ -0,0 +1,191 @@
+# What is MTLS?
+You’ve probably heard about TLS. It is a security protocol that sits between the transport and application layers (in the TCP/IP model) and essentially encrypts application-layer traffic. This encryption provides a guarantee to the client that:
+
+- The server is who they say they are.
+- Its communication with the server is confidential and cannot be spied on by a third party.
+- Data received is untouched/unmodified along the way.
+
+The regular TLS achieves all three above with a caveat - it provides no guarantee to the server that the client is who they say they are. As a result, verifying the client is done by the application using techniques like basic authentication (using usernames and passwords), Bearer/Access Tokens, or *mutual TLS*.
+
+## Like TLS, but mutual
+
+Mutual TLS (MTLS for short) is a part of the TLS standard. It extends the regular TLS by demanding a client-to-server authentication in addition to the default server-to-client one. To see how this extension happens, let’s summarily go over the steps involved in a typical TLS interaction (a.k.a a TLS handshake).
```

Comment:
This is good high level introduction. For our audience, I'd prefer to dig a bit deeper in actual mechanics of the certificates, exchanges, messages and how they provide the guarantees and prevent MITM

Reply:
For example, how would server know that client is in possession of the private key that matches the certificate and how exactly server trusts the cert? These are the key details we'd have to explain to the reader